pkg/cluster/internal/create/actions/kubeadminit/init.go

```diff
@@ -114,33 +115,81 @@ func (a *action) Execute(ctx *actions.ActionContext) error {
+
+type genericYaml map[string]interface{}
 
 // writeKubeConfig writes a fixed KUBECONFIG to dest
 // this should only be called on a control plane node
-// While copyng to the host machine the control plane address
+// While copying to the host machine the control plane address
 // is replaced with local host and the control plane port with
 // a randomly generated port reserved during node creation.
-func writeKubeConfig(n *nodes.Node, dest string, hostPort int32) error {
+// We also make the user reference id unique to allow to use
+// multiple kubeconfig files in the KUBECONFIG variable
+func writeKubeConfig(n *nodes.Node, dest string, hostPort int32, userSuffix string) error {
 	cmd := n.Command("cat", "/etc/kubernetes/admin.conf")
-	lines, err := exec.CombinedOutputLines(cmd)
+	buff, err := exec.Output(cmd)
 	if err != nil {
 		return errors.Wrap(err, "failed to get kubeconfig from node")
 	}
 
-	// fix the config file, swapping out the server for the forwarded localhost:port
-	var buff bytes.Buffer
-	for _, line := range lines {
-		match := serverAddressRE.FindStringSubmatch(line)
-		if len(match) > 1 {
-			line = fmt.Sprintf("%s https://localhost:%d", match[1], hostPort)
+	var config genericYaml
+	if err = yaml.Unmarshal(buff, &config); err != nil {
+		return errors.Wrap(err, "failed to parse kubeconfig file")
+	}
+
+	// Swap out the server for the forwarded localhost:port
+	cluster := config["clusters"].([]interface{})[0].(map[interface{}]interface{})
+	for k, v := range cluster {
+		if k == "cluster" {
+			clusterMap := v.(map[interface{}]interface{})
+			for k, v = range clusterMap {
+				if k == "server" {
+					clusterMap[k] = fmt.Sprintf("https://localhost:%d", hostPort)
+					break
+				}
+			}
+			break
 		}
-		buff.WriteString(line)
-		buff.WriteString("\n")
+	}
+
+	// Must make the user reference id unique to our cluster to allow
+	// for using the kubeconfig file of multiple clusters at the same
+	// time in the KUBECONFIG variable.
+
+	// Add a suffix to the user reference id in the context section
+	// which is in the "contexts[0].context.user" field.
+	context := config["contexts"].([]interface{})[0].(map[interface{}]interface{})
+	var newUserName, oldUserName string
+	for k, v := range context {
+		if k == "context" {
+			contextMap := v.(map[interface{}]interface{})
+			for k, v = range contextMap {
+				if k == "user" {
+					oldUserName = fmt.Sprintf("%s", v)
+					newUserName = fmt.Sprintf("%s-%s", oldUserName, userSuffix)
+					contextMap[k] = newUserName
+					break
+				}
+			}
+			break
+		}
+	}
+
+	// Use the new user reference id in the users section
+	// which is in the "users[0].name" field.
+	// In the same loop, add the 'username' field in the "users[0].user" section.
+	user := config["users"].([]interface{})[0].(map[interface{}]interface{})
+	for k := range user {
+		if k == "name" {
+			user[k] = newUserName
+		} else if k == "user" {
```

Comment:
Is this for backward compatibility? The current config looks like this:

```
(MoeLove) ➜  kind git:(using-network) ✗ kubectl config view 
apiVersion: v1
clusters:
- cluster:
    certificate-authority-data: DATA+OMITTED
    server: https://localhost:45965
  name: moe
contexts:
- context:
    cluster: moe
    user: kubernetes-admin
  name: kubernetes-admin@moe
current-context: kubernetes-admin@moe
kind: Config
preferences: {}
users:
- name: kubernetes-admin
  user:
    client-certificate-data: REDACTED
    client-key-data: REDACTED

```

Reply:
Which part?

As I understand it, the field `contexts[0].context.user` refers to a `user` defined in this file in the `users[]` section.  So, this PR makes the `user` unique and changes it in the two sections: `contexts[0].context.user` and `users[].name`.

Finally, the username within the cluster is taken by default to be `users[].name` but since we changed it, we need to tell what the real username is.  That is why we must insert the field `username`.

There is not backwards-compatibility issue to worry about, as far as I can tell, although I am no expert in kubeconfig definitions.
However, the real username in the cluster is still `kubernetes-admin`, which used to be the